src/util/lazyWithRetry.tsx

```diff
@@ -0,0 +1,62 @@
+import { lazy } from "react";
+import { delay } from "./helpers";
+
+// This function is a wrapper around React.lazy that will retry the import
+// keeping the function type signature the same as React.lazy
+// using "any" type here allows for components with any type interface to be imported
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const lazyWithRetry = <T extends React.ComponentType<any>>(
+  importFunction: () => Promise<{ default: T }>,
+  retries = 3,
+  delayTime = 1000,
+) => {
+  const retryImport = async () => {
+    try {
+      const module = await importFunction();
+      return module;
+    } catch (error) {
+      let lastError = error as Error;
+      for (let attempt = 0; attempt < retries; attempt++) {
+        await delay(delayTime * attempt);
+
+        // React.lazy unfortunately caches network response for fetching modules
+        // this is true even if a module fails to load
+        // if we simply just retry the import, React.lazy will return the same error response from its cache
+        // to work around this, we can invalidate the cache by adding a dynamic query parameter to the module URL
+        // the original module url can be potentially found in the error message
+        // HOWEVER, this is not a guaranteed solution since the error message may not contain the module URL depending on the browser used
+        // therefore, this is a best effort attempt, worst case scenario, the error will be thrown again causing the error boundary to render
+        const protocol = window.location.protocol;
```

Comment:
Probably safer to use the `window.location.origin`, as the protocol might be used in the error message itself. The origin is at least a bit more specific.

Reply:
nice! done :+1: